flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSource.java

```diff
@@ -88,7 +92,16 @@ private FileSystemTableSource(
 	}
 
 	@Override
-	public ScanRuntimeProvider getScanRuntimeProvider(ScanContext runtimeProviderContext) {
+	public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
+		Optional<BulkDecodingFormat<RowData>> bulkDecodingFormat = discoverBulkDecodingFormat();
+
+		if (!partitionKeys.isEmpty() && getOrFetchPartitions().isEmpty()) {
+			// When this table has no partition, just return a empty source.
+			return InputFormatProvider.of(new CollectionInputFormat<>(new ArrayList<>(), null));
+		} else if (bulkDecodingFormat.isPresent()) {
+			return SourceProvider.of(createBulkFormatSource(bulkDecodingFormat.get(), scanContext));
+		}
+
 		return new DataStreamScanProvider() {
 			@Override
 			public DataStream<RowData> produceDataStream(StreamExecutionEnvironment execEnv) {
```

Comment:
Why we should avoid using `ContinuousFileMonitoringFunction` here? and why not return `SourceFunctionProvider` of `InputFormatSourceFunction` directly?

Reply:
The `ContinuousFileMonitoringFunction` can not accept multiple paths. Default `StreamEnv.createInput` will create continuous function.

Reply:
Then can we return `SourceFunctionProvider.of(InputFormatSourceFunction)` instead of `DataStreamScanProvider` here?

Reply:
Indeed, we can!